.github/workflows/ci.yml

```diff
@@ -8,9 +8,207 @@ on:
+          tags: linux-debian-wine-image
+          push: false
+          load: false
+          cache-from: type=gha
+          cache-to: type=gha,mode=min
+
+  mingw_debian:
+    name: ${{ matrix.configuration.job_name }}
+    runs-on: ubuntu-latest
+    needs: docker_wine_cache
+
+    env:
+      WRAPPER_CMD: 'wine'
+      WITH_VALGRIND: 'no'
+      ECDH: 'yes'
+      RECOVERY: 'yes'
+      SCHNORRSIG: 'yes'
+      CTIMETESTS: 'no'
+
+    strategy:
+      fail-fast: false
+      matrix:
+        configuration:
+          - job_name: 'x86_64 (mingw32-w64): Windows (Debian stable, Wine)'
+            env_vars:
+              HOST: 'x86_64-w64-mingw32'
+          - job_name: 'i686 (mingw32-w64): Windows (Debian stable, Wine)'
+            env_vars:
+              HOST: 'i686-w64-mingw32'
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: CI script
+        env: ${{ matrix.configuration.env_vars }}
+        uses: ./.github/actions/run-in-docker-action
+        with:
+          dockerfile: ./ci/linux-debian-wine.Dockerfile
+          tag: linux-debian-wine-image
+          command: >
+            git config --global --add safe.directory ${{ github.workspace }} &&
+            ./ci/ci.sh
+
+      - run: cat tests.log || true
+        if: ${{ always() }}
+      - run: cat noverify_tests.log || true
+        if: ${{ always() }}
+      - run: cat exhaustive_tests.log || true
+        if: ${{ always() }}
+      - run: cat ctime_tests.log || true
+        if: ${{ always() }}
+      - run: cat bench.log || true
+        if: ${{ always() }}
+      - run: cat config.log || true
+        if: ${{ always() }}
+      - run: cat test_env.log || true
+        if: ${{ always() }}
+      - name: CI env
+        run: env
+        if: ${{ always() }}
+
+  msvc_debian:
+    name: ${{ matrix.configuration.job_name }}
+    runs-on: ubuntu-latest
+    needs: docker_wine_cache
+
+    env:
+      WRAPPER_CMD: 'wine'
+      WERROR_CFLAGS: '-WX'
+      WITH_VALGRIND: 'no'
+      ECDH: 'yes'
+      RECOVERY: 'yes'
+      EXPERIMENTAL: 'yes'
+      SCHNORRSIG: 'yes'
+      ELLSWIFT: 'yes'
+      CTIMETESTS: 'no'
+      # Use a MinGW-w64 host to tell ./configure we're building for Windows.
+      # This will detect some MinGW-w64 tools but then make will need only
+      # the MSVC tools CC, AR and NM as specified below.
+      CC: '/opt/msvc/bin/x64/cl'
+      AR: '/opt/msvc/bin/x64/lib'
+      NM: '/opt/msvc/bin/x64/dumpbin -symbols -headers'
+      # Set non-essential options that affect the CLI messages here.
+      # (They depend on the user's taste, so we don't want to set them automatically in configure.ac.)
+      CFLAGS: '-nologo -diagnostics:caret'
+      LDFLAGS: '-Xlinker -Xlinker -Xlinker -nologo'
+
+    strategy:
+      fail-fast: false
+      matrix:
+        configuration:
+          - job_name: 'x86_64 (MSVC): Windows (Debian stable, Wine)'
+            env_vars:
+              HOST: 'x86_64-w64-mingw32'
```

Comment:
Is there a reason why you moved this down to `configurations` as compared to the previous version of the PR?

Reply:
> Is there a reason why you moved this down to `configurations` as compared to the previous version of the PR?

Yes. The reason is to do not leave `matrix.configuration.env_vars` empty. Otherwise, `env: ${{ matrix.configuration.env_vars }}` fails.

Reply:
Oh okay, hm. What's the error message? (Just curious, perhaps I have an idea ...)

Reply:
Have you tried `env_vars: {}` ? It's the empty YAML dict, see https://stackoverflow.com/questions/33510094/syntax-for-empty-dictionary-in-yaml

Reply:
> Have you tried `env_vars: {}` ? It's the empty YAML dict, see [stackoverflow.com/questions/33510094/syntax-for-empty-dictionary-in-yaml](https://stackoverflow.com/questions/33510094/syntax-for-empty-dictionary-in-yaml)

Thank you! It [works](https://github.com/bitcoin-core/secp256k1/pull/1398#issuecomment-1675902686).